anomalib/utils/callbacks/save_to_csv.py

```diff
@@ -48,4 +48,8 @@ def on_test_epoch_end(self, _trainer: Trainer, pl_module: AnomalyModule) -> None
                 "wrong_prediction": np.logical_xor(results.true_labels, results.pred_labels).astype(int),
             }
         )
-        data_frame.to_csv(Path(pl_module.hparams.project.path) / "results.csv")
+
+        if trainer.log_dir is not None:
+            data_frame.to_csv(Path(trainer.log_dir) / "results.csv")
+        else:
+            raise ValueError("trainer.log_dir does not exist to save the results.")
```

Comment:
Should we maybe make the directory instead of throwing an error? Since this is cone at the end of training, we might have to train again. This might be inconvenient for algorithms like stfpm, cflow.

Reply:
I'll probably address this in another pr, which will be kind of interconnected